enarx-keep-sev-shim/src/frame_allocator.rs

```diff
@@ -0,0 +1,238 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! The global FrameAllocator
+use crate::BOOT_INFO;
+use memory::Page as Page4KiB;
```

Comment:
I think it makes sense to move Page2MiB into memory and refactor Page into Page4KiB, though I imagine that would balloon the PR quite a bit.

Reply:
@connorkuehl thoughts?

Reply:
I'm not opposed to it, but I'm also happy to leave it as is as an implementation detail that the VM/SEV shim worries about; at least until the VM/SEV shim is not the only component that needs such functionality.